### ISSUE TITLE
ContentPermissionsPart does not seem to work as expected

### ISSUE THREAD

**Issue by MatteoPiovanelli-Laser** [OrchardCMS/Orchard#7460]
I am having some issues with how the ContentPermissionsPart works.

- (minor) After I add a ContentPermissionsPart to a ContentType, in the section of the TypeEditor under "Only show those roles in the editor:" I uncheck all roles. When I go to the Editor view for a ContentItem, I see no role whose permissions I can change (correctly), but I still see the table headers for the permissions I may be able to change. I'd rather not have the table being drawn at all if it will not have any line.
 - The Orchard.ContentPermissions feature creates a GrantOwnPermission Permission, that is by default assigned to the "Author" role. However this is not used anywhere: the only check that is performed in the drivers (and in the TypePartEditor) is on GrantPermission. In the Editors, I would either check whether the current user is the owner of the ContenItem and verify the GrantOwnPermission (preferably), or remove this permission entirely, since it's not used anyway (could this cause issues?)
 - The ContentPermissionsPart does not seem to work. I mean, if I change permission settings for a Role/ContentType combination in the ContentType's TypeEditor, this is not reflected in the Editor for a ContentItem. This next image shows the way things are set in Page's TypeEditor:
![image](https://cloud.githubusercontent.com/assets/19283945/20889071/e0f1ee44-bb01-11e6-8485-5f2bc4ebcec6.png)
Note that I set things so that the Author role can "Edit any content". Even though I saved the settings the "Default value" checkbox stays unchecked. This happens with the type being either Securable or not. On top of that, the role Author cannot "Publish own content".
This shows the part in a ContentItem of type Page as seen by the admin (SiteOwner)
![image](https://cloud.githubusercontent.com/assets/19283945/20889114/0d2bff72-bb02-11e6-9f35-7fd11b1ca8bf.png)
As you can see, the Author role cannot "Edit this item" even though I would expect it to be able to.
Next, a list of content items of type Page, as seen by the user "matteo123", who has the Author role:
![image](https://cloud.githubusercontent.com/assets/19283945/20889271/9f3621f4-bb02-11e6-8d18-49a7c9edeb2c.png)
The first two items in the list have had the "Enable Content Item access control" checkbox checked by the admin. The last two items have not had that checkbox checked, so they are not protected. From the point of view of this user (again, matteo123 has the Author role) they appear identical. I was testing here whether the setting for the ContentItem was affecting somehow how the ContentType setting was being applied.
The third item in the list was created by the matteo123 Author user, so the user can do everything with it. Even Publish/Unpublish, even though I removed that permission in the Type settings. If I want to enable the "Content Item access control" on that item, I have to do it using the admin user, and by default all the permissions are disabled.
If in that ContentItem's editor I set all the "own" permissions (the second row), matteo123 again has full control of the item.
However, if instead I set all the "this item" permissions (the first row), and not any of the others, matteo123 again cannot do anything with the item (except view), even though it has the Author role.
On the other hand, if I set all the "this item" permissions for anther item, one that matteo123 does not own, then matteo123 can again do anything with it.

To summarize:
 - It looks like the permissions set in the TypeEditor for the ContentPermissionsPart are alway being ignored, or overridden by some other level of permission.
 - At the ContentItem Editor level:
   - If I don't "Enable Content Item access control", the ContentItem can be interacted with normally, as per the standard user permissions.
   - If I "Enable Content Item access control":
     - The "this item" permissions allow control for anyone (with the corresponding role) except the ContentItem's owner.
     - The "own" permissions allow control for the ContentItem's owner (as expected).

**Comment by sebastienros** [comment]
I will test the feature to ensure it works correctly, some of you statements are based on missunderstanding.
For instance the left column (disabled) of the checkboxes is to represent what the permission would be without this part, not "with this part". So it should not change when you check/uncheck custom permissions. The value you put in the part will always override any other logic. This is a special case in orchard where you can "remove" a permission. No other component can do that.
